sound/soc/sof/topology.c

```diff
@@ -2189,12 +2189,6 @@ static int sof_process_load(struct snd_soc_component *scomp, int index,
 	int ret;
 	int i;
 
```

Comment:
Code change looks good, but the commit message is hard to understand. How about?

```
ASoC: SOF: topology: make process type optional

As components can be now identified with a UUID based mechanism, the process
type is no longer required. New DSP components will set the process type
field to SOF_PROCESS_NONE. Modify the load time check for process
type to reflect this.
```

Reply:
The commit message says PROCESS_NONE, but based on the commit do you mean COMP_NONE?

Reply:
@cujomalainey I agree it is confusing. The FW actually uses the comp_type to determine which component driver to load. So PROCESS_NONE in the topology tokens equates to COMP_NONE.

Reply:
@cujomalainey @ranj063 yes, you can actually not specify the comp_type in topology as the default value will be PROCESS_NONE and COMP_NONE.

Reply:
> Code change looks good, but the commit message is hard to understand. How about?
> 
> ```
> ASoC: SOF: topology: make process type optional
> 
> As components can be now identified with a UUID based mechanism, the process
> type is no longer required. New DSP components will set the process type
> field to SOF_PROCESS_NONE. Modify the load time check for process
> type to reflect this.
> ```

Sounds good, thanks, let me change it.

Reply:
@keyonjie Hmm, this is indeed a bit confusing to have SOF_PROCESS_NONE in commit but SOF_COMP_NONE in code. How about one more adjustment:
``` 
As components can be now identified with a UUID based mechanism, the process
type is no longer required. For new DSP components, process and its component type 
can be set to SOF_PROCESS_NONE and SOF_COMP_NONE. Allow this combination in
topology load.
```